calicoctl/commands/resources.go

```diff
@@ -304,3 +249,51 @@ func executeResourceAction(args map[string]interface{}, client client.Interface,
 
 	return []runtime.Object{resOut}, err
 }
+
+// handleNamespace fills in the namespace information in the resource (if required),
+// and validates the namespace depending on whether or not a namespace should be
+// provided based on the resource kind.
+func handleNamespace(resource runtime.Object, args map[string]interface{}) (resourcemgr.ResourceObject, string, bool, error) {
+	allNs := argutils.ArgBoolOrFalse(args, "--all-namespaces")
+	namespace := argutils.ArgStringOrBlank(args, "--namespace")
+
+	// Check if --namespace and --all-namespaces flags are used together.
+	if allNs && namespace != "" {
+		return nil, "", false, fmt.Errorf("cannot specify a namespace with --all-namespace flag")
+	}
+
+	// If --all-namespaces is used then we must set namespace to "" so
+	// list operation can list resources from all the namespaces. Otherwise for
+	// the rest of the operations, we set the namespace to "default" if not specified.
+	if allNs {
+		namespace = ""
+	} else if namespace == "" {
+		namespace = "default"
+	}
+
+	if resourcemgr.IsNamespaced(resource) {
```

Comment:
I'm not we should have static methods for things like IsNamespaced.  Really we should be looking up the resource helper and using an accessor method on that to determine IsNamespaced().  I realize that means you have a little extra code, but it keeps things nicely encapsulated.  Besides, you can always pass the resource manager into this function since you have it to have when you call this function.

Reply:
Or alternatively we only have static methods and hide the actual resource helper lookup.

Reply:
WDYT?